client/my-sites/plans/jetpack-plans/utils.ts

```diff
@@ -245,26 +244,34 @@ export function slugIsFeaturedProduct( productSlug: string ): boolean {
 }
 
 /**
- * Gets a price in a set of price tiers.
+ * Gets a starting price in a set of price tiers.
  *
  * @param tiers A range of tiered pricing.
- * @param tierKey A key in the tiered pricing object.
- * @param units Optional. Number of units to use when dealing with variable pricing.
- * @returns {number|null} The amount it costs or null.
+ * @param units Number of units to use when dealing with variable pricing.
+ * @returns {string|null} The amount it costs or null if there is no matching tier.
  */
-export function getPriceTier(
-	tiers: PriceTiers,
-	tierKey: keyof PriceTiers,
-	units = 1
-): number | null {
-	if ( ! ( tierKey in tiers ) ) {
+function getPriceForUnitsInTier( tiers: PriceTierEntry[], units: number ): string | null {
+	const firstUnboundedTier = tiers.find( ( tier ) => ! tier.maximum_units );
+	let matchingTier = tiers.find( ( tier ) => {
+		if ( ! tier.maximum_units ) {
+			return false;
+		}
+		if ( units >= tier.minimum_units && units <= tier.maximum_units ) {
+			return true;
+		}
+		return false;
+	} );
+	if ( ! matchingTier && firstUnboundedTier && units >= firstUnboundedTier.minimum_units ) {
+		matchingTier = firstUnboundedTier;
+	}
+
+	if ( ! matchingTier ) {
 		return null;
 	}
-	const tier = tiers[ tierKey ];
-	if ( 'flat_price' in tier ) {
-		return tier.flat_price;
+	if ( matchingTier.flat_fee ) {
+		return matchingTier.flat_fee_monthly_display;
 	}
-	return tier.variable_price_per_unit * units;
+	return matchingTier.per_unit_fee_monthly_display;
```

Comment:
The logic in this method makes me uneasy because it seems to be trying to replicate the server-side price calculation on the client side.  I also think it might be doing it incorrectly in a couple ways (Shouldn't the per-unit-fee be multiplied by a number of units? And what happens if the tier has both a flat fee and a per-unit-fee?) -- although they don't impact the two places this is currently used.

I guess the previous code was sort of doing the same thing, but I don't think that was great either, and at least it was a simpler calculation.

Instead of passing all the internal details to the client and making the client responsible for calculating the price, I think we could just calculate the price on the server and then remove most of the code in Calypso?  I'll comment on D59709-code also.

Reply:
Honestly, this function only exists to get the `flat_fee_display` for the 1 and 101 tier, so we really don't need it at all. I will try replacing it with just accessing those two tier variables directly.

Reply:
I've modified the function to just return a tier that matches a particular quantity. Then we just use the `flat_fee_display` for that tier.

Reply:
That' still encoding a decent amount of knowledge of how the tiers work though (i.e. the fact that it only uses a `flat_fee_display` in the first place).  I guess if all this is named in a way that makes very clear it's specific to the Jetpack Search product it would be OK, but I think just returning the actual price from the server would likely be simpler.

Reply:
Good point. I've updated the schema to just include the units and prices.